src/render.c

```diff
@@ -148,6 +148,10 @@ void cmark_render_code_point(cmark_renderer *renderer, uint32_t c) {
   renderer->column += 1;
 }
 
+static inline int cmark_is_block(cmark_node *node) {
+  return node->type >= CMARK_NODE_FIRST_BLOCK && node->type <= CMARK_NODE_LAST_BLOCK;
+}
+
```

Comment:
I believe the naming conventions we've been using are that `cmark_*` is used for the functions in the public API.  So I think it may introduce confusion to name this function `cmark_is_block`.  Maybe use `is_block` or `S_is_block` (a conventioned that can be found in e.g. blocks.c.))

Reply:
Maybe a good reason to extract this into some decent public interface.

Reply:
Yes, I'd be in favor of (a) moving these to cmark.h/c, and making them public.
Maybe also (b) [re-organizing the enums the way cmark-gfm does](https://github.com/commonmark/cmark/pull/559#issuecomment-2295678470), to make the tests more efficient.

@nwellnhof can you think of any reason not to do these things?

Reply:
> Yes, I'd be in favor of (a) moving these to cmark.h/c, and making them public.

+1

> Maybe also (b) https://github.com/commonmark/cmark/pull/559#issuecomment-2295678470, to make the tests more efficient.

The test is efficient enough. I think this is more about extensibility. Technically, changing the enum values is an ABI break. This shouldn't be a problem as long as we continue to bump the soname with each release and people don't hardcode the constants. But I don't see a good reason to make the change as long as we don't add new node types. If `cmark_is_block` is made public, we should think about deprecating `CMARK_NODE_FIRST_BLOCK` and `CMARK_NODE_LAST_BLOCK`.

Reply:
Another function we should consider making public is `S_is_leaf` from `src/iterator.c`.

Reply:
OK, let's not change the enums at this point.  But let's add `cmark_is_block`, `cmark_is_inline`, and `cmark_is_leaf` to the public API in cmark.h.

@ioquatix do you want to do this as part of this PR, or should we do that separately and have you rebase?

Reply:
I think it's better to do it as a separate PR, and I'll rebase this one.